romancal/tests/test_requirements.py

```diff
@@ -0,0 +1,39 @@
+import json
+import re
+from pathlib import Path
+
+TEST_REQUIREMENTS_FILENAME = (
+    Path(__file__).parent.parent.parent / "test_requirements.json"
+)
+TEST_DIRECTORY = Path(__file__).parent.parent
+
+
+def test_requirements():
+    test_requirements_filename = TEST_REQUIREMENTS_FILENAME.expanduser().absolute()
+    test_directory = TEST_DIRECTORY.expanduser().absolute()
+
+    with open(test_requirements_filename) as test_requirements_file:
+        requirements = json.load(test_requirements_file)
+
+    required_tests = sorted(
+        {
+            test
+            for requirement_tests in requirements.values()
+            for test in requirement_tests
+        }
+    )
+
+    existing_tests = []
+    test_regex = re.compile(r"def (test_[^\(]+)\(.*\):")
```

Comment:
Can we make pytest do this work for us?
https://docs.pytest.org/en/7.1.x/example/special.html

Reply:
oh, interesting!

Reply:
Thanks for giving it a whirl. I poked at it a bit and wasn't able to get the linked fixture example to work but did get something working using a pytest hook: https://github.com/zacharyburnett/romancal/pull/3

Reply:
this seems to work well! from local testing it doesn't work when not all tests are loaded (i.e. `pytest romancal/tests/test_requirements.py`) but I don't think that is a usual test case

Reply:
Good point. If we do want to support that usage I think the one you originally have in this PR will be needed. I'm good with either one.